src/main/java/com/github/packageurl/PackageURL.java

```diff
@@ -543,23 +532,14 @@ private static String toLowerCase(String s) {
         return new String(chars);
     }
 
-    /**
-     * Optionally decodes a String, if it's encoded. If String is not encoded,
-     * method will return the original input value.
-     *
-     * @param input the value String to decode
-     * @return a decoded String
-     */
-    private String percentDecode(final String input) {
-        return uriDecode(input);
-    }
-
-    public static String uriDecode(String source) {
+    public static String percentDecode(final String source) {
```

Comment:
I don't know why somebody would be calling this method, but renaming the public method is technically a breaking change.

Reply:
Oh great, that was the public method. I had it flipped. I changed it back.

Either:

1. Keep both, but make percent public and deprecate the other one.
2. Remove the percent, and just leave uri

Reply:
IMHO, `uriDecode` should not be part of this library's **public** API. Therefore I would:

- deprecated `uriDecode`,
- make `percentDecode` private.